app/models/search.rb

```diff
@@ -16,6 +16,7 @@ def self.where_terms(string)
         "#{clean_str}:A"
       elsif /\A\d+\/\d+\z/.match?(str) then "#{str}:A"
       elsif /\A\d+{6}\z/.match?(str) then "#{str}:*"
+      elsif /[&|!()*]/.match?(str) then "'''#{str.tr('/', '-')}''':B" # Escape special tsquery characters
```

Comment:
Rather than escape if there are special characters and not if there aren't, would it not make more sense to just escape every time? That way:

1) if there are other special characters we forgot about, they'll be automatically escaped
2) we reduce the possible edge cases. E.g. here you've gone for ":B" instead of ":*B", and that has consequences for doing partial matching, so you've created an edge case where if someone includes a partial match of a defendant name and one of these special characters the system will behave differently to if they  do just a partial search or just a special character.

So can I suggest just putting:

```
else
   "'''#{str.tr('/', '-')}''':*B"
end
```

Reply:
I tried that in testing and it failed so I assumed you couldn't partial match with this.

Odd, okay changed.